src/components/DocumentationTopic/PrimaryContent/DeclarationSource.vue

```diff
@@ -52,6 +51,111 @@ export default {
+    //
+    // @param {Array} tokens The original syntax tokens.
+    //   See `DeclarationToken.props`
+    // @return {Array} A formatted version of the original tokens.
+    //   See `DeclarationToken.props`
+    formattedSwiftTokens: ({ tokens }) => {
+      const numSpacesForIndent = 4; // maybe this could be configurable in the future
+      const indent = ' '.repeat(numSpacesForIndent);
+      let indentedParams = false;
+      const newTokens = [];
+      let i = 0;
+      let j = 1;
+      let openParenTokenIndex = null;
+      let openParenCharIndex = null;
+      let closeParenTokenIndex = null;
+      let closeParenCharIndex = null;
+      let numUnclosedParens = 0;
+
+      // loop through every declaration token
+      while (i < tokens.length) {
+        // keep track of the current token and the next one (if any)
+        const token = tokens[i];
+        const newToken = { ...token };
+        const nextToken = j < tokens.length ? tokens[j] : undefined;
+
+        // loop through the token text to look for "(" and ")" characters
+        // eslint-disable-next-line no-plusplus
+        for (let k = 0; k < (token.text || '').length; k++) {
+          if (token.text.charAt(k) === '(') {
+            numUnclosedParens += 1;
+            // keep track of the token/character position of the first "("
+            if (openParenCharIndex == null) {
+              openParenCharIndex = k;
+              openParenTokenIndex = i;
+            }
+          }
+
+          if (token.text.charAt(k) === ')') {
+            numUnclosedParens -= 1;
+            // if this ")" balances out the number of "(" characters that have
+            // been seen, this is the one that pairs up with the first one
+            if (openParenTokenIndex !== null && numUnclosedParens === 0) {
+              closeParenCharIndex = k;
+              closeParenTokenIndex = i;
+            }
+          }
+        }
+
+        // if we find some text ending with ", " and the next token is the start
+        // of a new param, update this token text to replace the space with a
+        // newline followed by 4 spaces
+        if (token.text && token.text.endsWith(', ')
+          && nextToken && nextToken.kind === TokenKind.externalParam) {
+          newToken.text = `${token.text.trimEnd()}\n${indent}`;
+          indentedParams = true;
+        }
+
+        newTokens.push(newToken);
+        i += 1;
+        j += 1;
+      }
+
+      // if we indented some params, we want to find the opening "(" symbol
+      // and add a newline and 4 spaces to the end of it, breaking the first
+      // param onto its own line
+      if (indentedParams && openParenTokenIndex !== null) {
+        const originalText = newTokens[openParenTokenIndex].text;
+        const begin = originalText.slice(0, openParenCharIndex);
+        const end = originalText.slice(openParenCharIndex);
+        const newText = `${begin}${end}\n${indent}`;
```

Comment:
Here wont it work to just append `/n${indent}`, why do we need to break up the text, then put it back together?

Reply:
I probably was just thinking too hard. I'll look at making that update haha 😅

Reply:
a20c63c9d51414407412887a076ada75e3f051ff 😆